File: src/android/kumakore_sample/src/com/kumakore/sample/test/TestAchievements.java
package com.kumakore.sample.test;

import android.util.Log;


import com.kumakore.ActionAchievementGetApp;
import com.kumakore.ActionAchievementGetUser;
import com.kumakore.ActionAchievementSetUser;
import com.kumakore.AppAchievement;
import com.kumakore.KumakoreApp;
import com.kumakore.UserAchievement;

public class TestAchievements extends TestBase {
	private static final String TAG = TestAchievements.class.getName();

	public TestAchievements(KumakoreApp app) {
		super(app);
	}

	@Override
	protected void onRun() {
		testAppAchievements();
		testUserAchievements();
	}

	public void testAppAchievements() {

		// ## example 9 : get app achievements ##
		app().getAchievements().get().async(new ActionAchievementGetApp.IKumakore() {
			@Override
			public void onActionAppAchievmentListGet(ActionAchievementGetApp action) {

				Log.i(TAG, "Achievements size " + app().getAchievements().size());
				for (AppAchievement achievement : app().getAchievements().values()) {
					Log.i(TAG, achievement.getName() + "  " + achievement.getDescription());
				}
			}

		});
	}

	public void testUserAchievements() {

		// ## example 7 : get user achievements ##
		app().getUser().achievements().get().async(new ActionAchievementGetUser.IKumakore() {

			@Override
			public void onActionAchievementGetUser(ActionAchievementGetUser action) {
				Log.i(TAG, "User Achievements size " + app().getUser().achievements().size());
				for (UserAchievement achievement : app().getUser().achievements().values()) {
					Log.i(TAG, achievement.getName() + "  " + achievement.getProgress());
				}
				testChangeProgress();
			}
		});

		
	}
	
	private void testChangeProgress() {
		UserAchievement achievement = app().getUser().achievements().get(0);
		Log.i(TAG, "Progress is " + achievement.getProgress());
		
		app().getUser().achievements().set(achievement.getName(), achievement.getProgress() + 1)
				.async(new ActionAchievementSetUser.IKumakore() {

					@Override
					public void onActionUserAchievmentSet(ActionAchievementSetUser action) {
						for (UserAchievement achievement : app().getUser().achievements().values()) {

							Log.i(TAG, achievement.getName() + "  " + achievement.getProgress());
						}
					}
				});
	}  
}

File: deliveries/v0.9.0/bin/unity3d/Assets/Plugins/com.kumakore/test/scripts/TestAchievements.cs
using System;
using com.kumakore;
using NUnit.Framework;
using System.Threading;

namespace com.kumakore.test
{
    [TestFixture]
	public class TestAchievements : TestBase
    {
		private static readonly String ACHIEVEMENT_1 = "ach_1"; // valid achievement name for app
		private static readonly int NUMBER_OF_ACHIEVEMENTS = 2; // number of achievements on app
		
		// run testcases from Unity scene
		public static void All()
        {
            TestAchievements app = new TestAchievements();
            app.setup();
			
			// app achievements
			app.SyncGetAchievementList();
			app.AsyncGetAchievementList();
			
			// user achievements
			app.SyncGetUserAchievementList();
			app.AsyncGetUserAchievementList();
			app.SyncSetUserAchievement();
			app.AsyncSetUserAchievement();
        }
		
		[TestFixtureSetUp]
		public override void setup ()
		{
			base.setup ();
			app1 ().signin(TEST_1,PASSWORD).sync ();
		}
		
		[Test]
        public void SyncGetAchievementList()
        {
            app1().getAchievements().get().sync(delegate(ActionAchievementGetApp action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.AreEqual (NUMBER_OF_ACHIEVEMENTS, app1().getAchievements().Count);
            });
        }

        [Test]
        public void AsyncGetAchievementList()
        {
            app1().getAchievements().get().async(delegate(ActionAchievementGetApp action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				Assert.AreEqual (NUMBER_OF_ACHIEVEMENTS, app1().getAchievements().Count);
                Release();
            });

            Wait();
        }
		
		[Test]
        public void SyncGetUserAchievementList()
        {
            app1().getUser().getAchievements().get().sync(delegate(ActionAchievementGetUser action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
            });
        }

        [Test]
        public void AsyncGetUserAchievementList()
        {
            app1().getUser().getAchievements().get().async(delegate(ActionAchievementGetUser action)
            {
                Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
                Release();
            });

            Wait();
        }
		
		[Test]
        public void SyncSetUserAchievement()
        {
			Double progress = 1.0;
			foreach(UserAchievement ach in app1().getUser().getAchievements().Values) {
				if(ACHIEVEMENT_1.Equals(ach.getName ())) progress += ach.getProgress();
			}
			app1().getUser().getAchievements().set (ACHIEVEMENT_1,progress).sync(delegate(ActionAchievementSetUser action)
            {
				Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				foreach(UserAchievement a in app1().getUser().getAchievements().Values) {
					if(a.getName () == ACHIEVEMENT_1) Assert.AreEqual(progress,a.getProgress());
				}
            });
        }

        [Test]
        public void AsyncSetUserAchievement()
        {
			Double progress = 1.0;
			foreach(UserAchievement ach in app1().getUser().getAchievements().Values) {
				if(ACHIEVEMENT_1.Equals(ach.getName ())) progress += ach.getProgress();
			}
			app1().getUser().getAchievements().set (ACHIEVEMENT_1,progress).async(delegate(ActionAchievementSetUser action)
            {
				Assert.AreEqual(StatusCodes.SUCCESS, action.getCode());
				foreach(UserAchievement a in app1().getUser().getAchievements().Values) {
					if(a.getName () == ACHIEVEMENT_1) Assert.AreEqual(progress,a.getProgress());
				}
				Release ();
            });
			Wait ();
        }
		
	}
}